src/core/render/compiler/link.js

```diff
@@ -1,23 +1,37 @@
 import { getAndRemoveConfig } from '../utils';
 import { isAbsolutePath } from '../../router/util';
 
-export const linkCompiler = ({ renderer, router, linkTarget, compilerClass }) =>
-  (renderer.link = (href, title = '', text) => {
+export const linkCompiler = ({
+  renderer,
+  router,
+  linkTarget,
+  compiler,
+  isSidebar = false,
+}) => {
+  const link = (href, title = '', text) => {
     let attrs = [];
     const { str, config } = getAndRemoveConfig(title);
 
     title = str;
 
     if (
       !isAbsolutePath(href) &&
-      !compilerClass._matchNotCompileLink(href) &&
+      !compiler._matchNotCompileLink(href) &&
       !config.ignore
     ) {
-      if (href === compilerClass.config.homepage) {
+      const sidebarAbsolutePath = compiler.config.sidebarAbsolutePath;
+      const basePath =
```

Comment:
To me the following lines has a weird formatting.

I would suggest doing this instead:
```suggestion
const basePath = typeof sidebarAbsolutePath === 'string'
          ? sidebarAbsolutePath
          : router.getBasePath();
const relativeTo = isSidebar && sidebarAbsolutePath 
          ? basePath 
          : router.getCurrentPath();
```